src/profile-logic/profile-data.js

```diff
@@ -1316,6 +1316,13 @@ export function filterThreadSamplesToRange(
     );
   }
 
+  if (samples.threadId) {
+    newSamples.threadId = samples.threadId.slice(
+      beginSampleIndex,
+      endSampleIndex
+    );
+  }
```

Comment:
The current tests don't exercize this, but when we'll start looking at changing the display of the marker chart depending on the source threads this will be used.